OpenDreamRuntime/Procs/Native/DreamProcNativeList.cs

```diff
@@ -123,27 +123,44 @@ internal static class DreamProcNativeList {
         [DreamProcParameter("Item1")]
         public static DreamValue NativeProc_Remove(NativeProc.Bundle bundle, DreamObject? src, DreamObject? usr) {
             DreamList list = (DreamList)src!;
-            bool itemRemoved = false;
+            return new DreamValue(ListRemove(list, bundle.Arguments) > 0 ? 1 : 0);
+        }
 
-            foreach (var argument in bundle.Arguments) {
+        [DreamProc("RemoveAll")]
+        [DreamProcParameter("Item1")]
+        public static DreamValue NativeProc_RemoveAll(NativeProc.Bundle bundle, DreamObject? src, DreamObject? usr) {
+            DreamList list = (DreamList)src!;
+            var totalRemoved = 0;
+            var removed = 0;
```

Comment:
The initial value is never used
```suggestion
            int removed;
```